File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    l = [0 for _ in range(2001)]
    l[1] = 1
    l[2] = 2
    for i in range(3, n+1):
        l[i] = (l[i-1] + l[i-2])%1234567
    return l[n]

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(int n) {
    long long answer = 0;
    vector<long long> v;
    v.push_back(0);
    v.push_back(1);
    v.push_back(2);
    for(int i=3; i<=n; i++)
        v.push_back((v[i-1] + v[i-2]) % 1234567);
    return v[n];
}


/*
1
11 2
111 21 12
1111 112 121 211 22
11111 1112 1121 1211 2111 122 212 221
1, 2, 3, 5, 8


*/